katdal/categorical.py

```diff
@@ -544,6 +582,9 @@ def sensor_to_categorical(sensor_timestamps, sensor_values, dump_midtimes, dump_
     sensor_timestamps = np.atleast_1d(sensor_timestamps)
     sensor_values = np.atleast_1d(sensor_values)
     dump_midtimes = np.atleast_1d(dump_midtimes)
+    # Check if sensor values are objects wrapped in HashableValueWrappers
+    wrapped_values = len(sensor_values) and isinstance(sensor_values[0],
+                                                       HashableValueWrapper)
```

Comment:
The ```sensor_values``` variable is desired to be an ndarray to simplify manipulation, and that would break other things if it contains unwrapped values. Rather wrap the incoming parameters... At least the greedy values can now be compared directly with the new ```ComparableArrayWrapper```.